lua/orgmode/org/mappings.lua

```diff
@@ -659,79 +659,83 @@ function OrgMappings:meta_return(suffix)
     if not listitem or listitem:type() ~= 'listitem' then
       return
     end
-    local line = vim.fn.getline(listitem:start() + 1)
-    local srow, _, end_row, end_col = listitem:range()
-    local is_multiline = (end_row - srow) > 1 or end_col == 0
-    -- For last item in file, ts grammar is not parsing the end column as 0
-    -- while in other cases end column is always 0
-    local is_last_item_in_file = end_col ~= 0
-    if not is_multiline or is_last_item_in_file then
-      end_row = end_row + 1
-    end
-    local range = {
-      start = { line = end_row, character = 0 },
-      ['end'] = { line = end_row, character = 0 },
-    }
-
-    local checkbox = line:match('^(%s*[%+%-%*])%s*%[[%sXx%-]?%]')
-    local plain_list = line:match('^%s*[%+%-%*]')
-    local indent, number_in_list, closer = line:match('^(%s*)(%d+)([%)%.])%s?')
-    local text_edits = config:respect_blank_before_new_entry({}, 'plain_list_item', {
+    return self:insert_item_below_this(listitem)
+  end
+end
+
+function OrgMappings:insert_item_below_this(listitem)
```

Comment:
Let's make this private and change the name:

```suggestion
---@private
---@param listitem OrgListitem
function OrgMappings:_insert_item_below(listitem)
```